samples/openapi3/client/petstore/dart2/petstore_client_lib/lib/auth/http_basic_auth.dart

```diff
@@ -11,14 +11,14 @@
 part of openapi.api;
 
 class HttpBasicAuth implements Authentication {
-  HttpBasicAuth(this.username, this.password);
+  HttpBasicAuth({this.username, this.password});
```

Comment:
same change here, this shouldnt be nullable or optional

Reply:
In this case, we could keep the code as-is (but make both properties non-nullable), but instead change `applyToParams()` to conditionally include a header if both are non-empty, what do you think?

Reply:
yea, sounds good. In general, not only this case, I think generated code shouldn't have null checks and we shouldn't decide what to put in place of a null object when we can move this responsibility on users of the API. They should decide if they want empty string, empty map/list etc. It removes responsibility and future proofs the API. If for some reason, it becomes necessary to change it, it will be easier to make this nullable, than not-nullable - wouldn't be a breaking change.

Reply:
Sounds good. I can make the necessary changes in around 2 hours, going for a long walk :)